core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1875,6 +1889,14 @@ class SocketServerTest {
         cachedCompletedReceives.update(super.completedReceives.asScala.toBuffer)
         cachedCompletedSends.update(super.completedSends.asScala)
         cachedDisconnected.update(super.disconnected.asScala.toBuffer)
+
+        val map: util.Map[String, NetworkReceive] = JTestUtils.fieldValue(this, classOf[Selector], "completedReceives")
+        cachedCompletedReceives.currentPollValues.foreach { receive =>
+          val channelOpt = Option(super.channel(receive.source)).orElse(Option(super.closingChannel(receive.source)))
+          channelOpt.foreach { channel => map.put(channel.id, receive) }
+        }
+        cachedCompletedSends.currentPollValues.foreach(super.completedSends.add)
+        cachedDisconnected.currentPollValues.foreach { case (id, state) => super.disconnected.put(id, state) }
```

Comment:
Can we add a comment explaining what we're trying to do here? It's not clear why we do (for example):

`cachedCompletedSends.update(super.completedSends.asScala)` followed by
`cachedCompletedSends.currentPollValues.foreach(super.completedSends.add)`

Reply:
I have moved the second line that updates current result into `update()`. They were done separately because each type uses slightly different format, but it is clearer if they are together. Added comments as well.